src/Adapter/Product/CommandHandler/UpdateProductOptionsHandler.php

```diff
@@ -0,0 +1,259 @@
+        }
+
+        if (null !== $command->getUpc()) {
+            $product->upc = $command->getUpc()->getValue();
+            $this->fieldsToUpdate['upc'] = true;
+        }
+    }
+
+    /**
+     * @param Product $product
+     * @param array|null $localizedTags
+     *
+     * @throws CannotUpdateProductException
+     * @throws ProductException
+     */
+    private function performUpdate(Product $product, ?array $localizedTags)
+    {
+        try {
+            if (false === $product->update()) {
+                throw new CannotUpdateProductException(
+                    sprintf(
+                        'Failed to update product #%s options',
+                        $product->id
+                    ),
+                    CannotUpdateProductException::FAILED_UPDATE_OPTIONS
+                );
+            }
+
+            $this->updateTags($product, $localizedTags);
+        } catch (PrestaShopException $e) {
+            throw new ProductException(
+                sprintf(
+                    'Error occurred during product #%s options update',
+                    $product->id
+                ),
+                0,
+                $e
+            );
+        }
+    }
+
+    /**
+     * Due to this method it is possible to partially update tags in separate languages.
+     *
+     * If all localizedTags array is empty, all previous tags will be deleted.
+     * If tags in some language are null, it will be skipped.
+     * If tags in some language has any values, all previous tags will be deleted and new ones inserted instead
+     * If tags in some language are empty, then all previous tags will be deleted and none inserted.
+     *
+     * @param Product $product
+     * @param array $localizedTags
+     *
+     * @throws CannotUpdateProductException
+     * @throws PrestaShopException
+     */
+    private function updateTags(Product $product, ?array $localizedTags)
+    {
+        // don't do anything with tags if its null. (It means it is a partial update and tags aren't changed)
+        if (null === $localizedTags) {
+            return;
+        }
+
+        $productId = (int) $product->id;
+
+        // delete all tags for product if array is empty
+        if (empty($localizedTags)) {
+            $this->deleteAllTagsForProduct($productId);
+
+            return;
+        }
+
+        foreach ($localizedTags as $langId => $tags) {
+            // don't do anything with this language tags if its null (equivalent to other partial updates)
+            if (null === $tags) {
+                continue;
+            }
+
+            // delete all this product tags for this lang
+            if (false === Tag::deleteTagsForProduct($productId, $langId)) {
+                throw new CannotUpdateProductException(
+                    sprintf('Failed to delete product #%s previous tags in lang #%s', $productId, $langId),
+                    CannotUpdateProductException::FAILED_UPDATE_OPTIONS
```

Comment:
Maybe a dedicated error code here would be useful to tell the difference with product update
```suggestion
                    CannotUpdateProductException::FAILED_UPDATE_TAGS
```

Reply:
Then it tells us, that maybe we should separate Tags from options command ?